### PR TITLE
Add mp-drop-menu-list-item

### PR DESCRIPTION
**Pull request by mduan**
image

### CODE REVIEW

File: lib/components/drop-menu-list-item/index.jade [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    'is-selectable': isSelectable,
+    'is-upper-case': isUpperCase,
+  }
+  on={click: !isSelectable ? e => e.stopPropagation() : null}
+)
+  if icon
+    .icon
+      svg-icon(attrs={icon})
+  .label
+    .primary-label(attrs={title: label ? label : ``})
```
Comment (tdumitrescu) [inline lib/components/drop-menu-list-item/index.jade]:
```js
label || ``
```

File: lib/components/drop-menu-list-item/index.jade [hunk 2]
```diff
@@ -0,0 +1,32 @@
+- const isSelectable = !isSelected || hasCaret || isSelectedSelectable
+.drop-menu-list-item(
+  class={
+    'is-selected': isSelected,
+    'is-selectable': isSelectable,
+    'is-upper-case': isUpperCase,
+  }
+  on={click: !isSelectable ? e => e.stopPropagation() : null}
+)
+  if icon
```
Comment (tdumitrescu) [inline lib/components/drop-menu-list-item/index.jade]: one of the issues you need to look out for here is menus where some items get icons and some don't; you don't want the no-icon ones displaced to the left. for those cases we've used the `empty` attr of `svg-icon` in existing menus

File: lib/components/drop-menu-list-item/index.jade [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    .icon
+      svg-icon(attrs={icon})
+  .label
+    .primary-label(attrs={title: label ? label : ``})
+      if labelParts
+        each labelPart, idx in labelParts
+          span.label-part(class={'is-match': idx % 2 === 0})= labelPart
```
Comment (tdumitrescu) [inline lib/components/drop-menu-list-item/index.jade]: i feel like there's already a jade mixin for match highlighting in common...

**Comment by mduan** [comment]
Superseded by https://github.com/mixpanel/mixpanel-common/pull/209

**Pull request closed without merging**